src/software/simulated_tests/non_terminating_validation_functions/ball_in_play_or_scored_validation.cpp

```diff
@@ -0,0 +1,16 @@
+#include "software/simulated_tests/non_terminating_validation_functions/ball_in_play_or_scored_validation.h"
+
+void ballInPlay(std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield)
+{
+    Point ball           = world_ptr->ball().position();
+    Rectangle inPlayRect = world_ptr->field().fieldLines();
+
+    if (contains(world_ptr->field().enemyGoal(), ball))
+    {
+        return;
+    }
+    if (!contains(inPlayRect, ball))
+    {
```

Comment:
minor suggestion: could make this more compact, by skipping the first if statement and then failing if the field doesn't contain the ball AND the ball isn't in the enemy goal.

Reply:
made it nicer